src/olympia/shelves/forms.py

```diff
@@ -38,25 +40,38 @@ def clean(self):
         if criteria is None:
             return
 
-        if endpoint == 'search':
+        if endpoint in (Shelf.Endpoints.SEARCH, Shelf.Endpoints.TAGS):
             if not criteria.startswith('?') or criteria.count('?') > 1:
                 raise forms.ValidationError('Check criteria field.')
-            params = criteria[1:].split('&')
-            if addon_type == amo.ADDON_EXTENSION and 'type=statictheme' in params:
+            params = dict(parse.parse_qsl(criteria.strip('?')))
+            if (
+                addon_type == amo.ADDON_EXTENSION
+                and params.get('type') == 'statictheme'
+            ):
                 raise forms.ValidationError(
                     'Use "Theme (Static)" in Addon type field for type=statictheme.'
                 )
             elif (
-                addon_type == amo.ADDON_STATICTHEME and 'type=statictheme' not in params
+                addon_type == amo.ADDON_STATICTHEME
+                and params.get('type') != 'statictheme'
             ):
                 raise forms.ValidationError(
                     'Check fields - for "Theme (Static)" addon type, use '
                     'type=statictheme. For non theme addons, use "Extension" in Addon '
                     'type field, not "Theme (Static)".'
                 )
+            if endpoint == Shelf.Endpoints.TAGS:
+                if 'tag' in params:
+                    raise forms.ValidationError(
+                        'Omit `tag` param for tags shelf - a random tag will be chosen.'
+                    )
+                params['tag'] = Tag.objects.first().tag_text
```

Comment:
What does this do?

Reply:
the code tries to call the API with the criteria set to validate it works, and normally we'd be calling the API with a random tag value.  To replicate this - in a consistent way - it sets the tag to the first Tag instance we have in the database.